modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/TriggersTest.kt

```diff
@@ -38,6 +44,180 @@ class TriggersTest {
+        val domainId = "wonderland".asDomainId()
+        val newAssetName = "token1"
+
+        // check DEFAULT_ASSET_ID quantity before trigger
+        val prevQuantity = QueryBuilder.findDomainById(domainId)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+            .let { query -> client.sendQuery(query) }
+            .accounts
+            .filter { it.key.name == ALICE_ACCOUNT_NAME }
+            .map { it.value.assets[DEFAULT_ASSET_ID] }
+            .map { (it?.value as AssetValue.Quantity).u32 }
+            .first()
+        assertEquals(100L, prevQuantity)
+
+        // register pre commit trigger
+        client.sendTransaction {
+            accountId = ALICE_ACCOUNT_ID
+            registerPreCommitTrigger(
+                triggerId,
+                listOf(Instructions.mintAsset(DEFAULT_ASSET_ID, 10L)),
+                Repeats.Indefinitely(),
+                ALICE_ACCOUNT_ID
+            )
+            executeTrigger(triggerId)
+            buildSigned(ALICE_KEYPAIR)
+        }.also {
+            Assertions.assertDoesNotThrow {
+                it.get(10, TimeUnit.SECONDS)
+            }
+        }
+
+        // check DEFAULT_ASSET_ID quantity after trigger is run
+        var newQuantity = QueryBuilder.findDomainById(domainId)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+            .let { query -> client.sendQuery(query) }
+            .accounts
+            .filter { it.key.name == ALICE_ACCOUNT_NAME }
+            .map { it.value.assets[DEFAULT_ASSET_ID] }
+            .map { (it?.value as AssetValue.Quantity).u32 }
+            .first()
+        assertEquals(110L, newQuantity)
+
+        // register new asset
+        // after that trigger should mint 10 more quantity to DEFAULT_ASSET_ID
+        createNewAsset(newAssetName)
+
+        // check DEFAULT_ASSET_ID quantity after trigger is run
+        newQuantity = QueryBuilder.findDomainById(domainId)
+            .account(ALICE_ACCOUNT_ID)
+            .buildSigned(ALICE_KEYPAIR)
+            .let { query -> client.sendQuery(query) }
+            .accounts
+            .filter { it.key.name == ALICE_ACCOUNT_NAME }
+            .map { it.value.assets[DEFAULT_ASSET_ID] }
+            .map { (it?.value as AssetValue.Quantity).u32 }
+            .first()
+        assertEquals(120L, newQuantity)
+
+        // transfer asset instruction just to test trigger
+        val bobAssetId = AliceAndBobEachHave100Xor.BOB_ASSET_ID
+        client.sendTransaction {
+            account(ALICE_ACCOUNT_ID)
+            transferAsset(DEFAULT_ASSET_ID, 100, bobAssetId)
+            buildSigned(ALICE_KEYPAIR)
+        }.also {
+            Assertions.assertDoesNotThrow {
+                it.get(10, TimeUnit.SECONDS)
+            }
+        }
+
+        // check DEFAULT_ASSET_ID quantity after trigger is run
+        newQuantity = QueryBuilder.findDomainById(domainId)
```

Comment:
It repeats 3 times, please consider create separate func or anonymous func like in test `InstructionsTest.check assets with type Fixed are properly minted and burned`